pyinfraformat/core/coord_utils.py

```diff
@@ -0,0 +1,538 @@
+    other_systems = {"HELSINKI": proj_helsinki, "ESPOO": proj_espoo, "PORVOO": proj_porvoo}
+
+    hole_copy = deepcopy(hole)
+    if not isinstance(hole, Hole):
+        raise ValueError("hole -parameter invalid")
+    if output_epsg not in epsg_names:
+        raise ValueError("Unkown or not implemented EPSG as output_epsg")
+    if (
+        hasattr(hole_copy, "fileheader")
+        and hasattr(hole_copy.fileheader, "KJ")
+        and "Coordinate system" in hole_copy.fileheader.KJ
+        and hasattr(hole_copy, "header")
+    ):
+        input_str = hole_copy.fileheader.KJ["Coordinate system"]
+        input_str = coord_string_fix(input_str)
+
+    else:
+        raise ValueError("Hole has no coordinate system")
+
+    if (
+        hasattr(hole_copy.header, "XY")
+        and "X" in hole_copy.header.XY
+        and "Y" in hole_copy.header.XY
+    ):
+        x, y = hole_copy.header.XY["X"], hole_copy.header.XY["Y"]
+        if not np.isfinite(x) or not np.isfinite(y):
+            raise ValueError("Coordinates are not finite")
+    else:
+        raise ValueError("Hole has no coordinates")
+
+    if input_str in epsg_systems:
+        input_epsg = epsg_systems[input_str][0]
+        if not epsg_systems[input_str][1]:
+            x, y = y, x
+    elif input_str in other_systems:
+        func = other_systems[input_str]
+        x, y, input_epsg = func(x, y)
+    else:
+        msg = "Unkown or not implemented EPSG in holes {}".format(input_str)
+        raise ValueError(msg)
+
+    key = (input_epsg, output_epsg)
+    if key in TRANSFORMERS:
+        transf = TRANSFORMERS[key]
+    else:
+        transf = Transformer.from_crs(input_epsg, output_epsg)
+        TRANSFORMERS[key] = transf
+    x, y = transf.transform(x, y)
+    if not epsg_names[output_epsg][1]:
+        hole_copy.header.XY["X"], hole_copy.header.XY["Y"] = y, x  # Note x-y change
+    else:
+        hole_copy.header.XY["X"], hole_copy.header.XY["Y"] = x, y
+
+    hole_copy.fileheader.KJ["Coordinate system"] = epsg_names[output_epsg][0]
+
+    if not output_height:
+        return hole_copy
+
+    key = (output_epsg, "EPSG:2393")
+    if key in TRANSFORMERS:
+        transf = TRANSFORMERS[key]
+    else:
+        transf = Transformer.from_crs(output_epsg, "EPSG:2393")
+        TRANSFORMERS[key] = transf
+    point = transf.transform(x, y)
+    try:
+        input_system = hole.fileheader["KJ"]["Height reference"]
+    except KeyError:
+        raise ValueError("Hole has no height system")
+    if input_system not in ["N43", "N60", "N2000"]:
+        raise ValueError("Hole has no unknown heigth system:", input_system)
+
+    diff = height_systems_diff(point, input_system, output_height)
+    hole_copy.header.XY["Z-start"] += round(float(diff), 3)
+    hole_copy.fileheader["KJ"]["Height reference"] = output_height
+    return hole_copy
+
+
+def project_holes(holes, output="EPSG:4326", check="Finland", output_height=False):
```

Comment:
```suggestion
def project_holes(holes, output="EPSG:3857", check="Finland", output_height=False):
```

Reply:
Tileproviders use Web Mercator (EPSG:3857), but the markers added to map are in WGS 84 (EPSG:4326). Our code will work wihout any mention of Web Mercator, folium does the work for us.